assets/js/admin/admin.js

```diff
@@ -0,0 +1,298 @@
+                    wpCookies.set(safeSvgCookie, '1', 'Session', '/', '', true);
+                }
+            });
+        } else {
+            // We are on the multi-file uploader, so we observe for new entries and if there is an SVG file, we optimize it.
+            const observer = new MutationObserver((mutations) => {
+                mutations.forEach((mutation) => {
+                    if (mutation.addedNodes.length > 0) {
+                        mutation.addedNodes.forEach((node) => {
+                            if (node.nodeType === Node.ELEMENT_NODE && node.matches('.media-item-wrapper')) {
+                                const image = node.querySelector('img');
+                                if (image?.src && image.src.endsWith('.svg')) {
+                                    fetch(image.src, {method: 'GET'})
+                                        .then((response) => response.text())
+                                        .then((response) => {
+                                            const params = ajaxUrlParams(image.src, response);
+                                            if (!params) {
+                                                return;
+                                            }
+                                            // Make an AJAX call to update the SVG file with the optimized contents.
+                                            ajaxUrl.search = new URLSearchParams(params);
+                                            fetch(ajaxUrl, {method: 'GET'})
+                                                .then((ajaxResponse) => ajaxResponse)
+                                        });
+                                }
+                            }
+                        });
+                    }
+                });
+            });
+            observer.observe(document.body, {childList: true, subtree: true});
+        }
+    }
+
+    /**
+     * If we are on the "Upload New Media" page (`wp-admin/media-new.php`), and the user has just uploaded an SVG file, optimize it.
+     */
+    if ('media-new.php' === context) {
+        const form = document.querySelector('#file-form');
+        optimizeFromUploadNewMedia(form);
+
+        // If the user switches between the browser uploader and the multi-file uploader, we need to re-run the optimization.
+        const observer = new MutationObserver(mutations => {
+            mutations.forEach(mutation => {
+                if (mutation.attributeName === 'class') {
+                    optimizeFromUploadNewMedia(form);
+                }
+            });
+        });
+        observer.observe(form, {attributes: true});
+    }
+
+    /**
+     * Hook into the WordPress Uploader and optimize the SVG.
+     */
+    if (wp.Uploader !== undefined) {
+        Object.assign(wp.Uploader.prototype, {
+            // Run on a successful upload.
+            success(attachment) {
+                const svgUrl = attachment?.attributes?.url;
```

Comment:
In the same data, you can get the ID of the image as `svgId = attachment?.attributes?.id` -- this may remove the need to call `attachment_url_to_postid` in the PHP, as it's quite an expensive function.

Reply:
Done.